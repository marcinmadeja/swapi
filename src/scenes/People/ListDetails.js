import React from 'react';
import PropTypes from 'prop-types';
import {
  Name,
  InfoItem,
} from 'components/SwapiList/styles';

const ListDetails = ({ details }) => {
  const { name, birth_year, gender } = details;

  return (
    <div>
      <Name>
        {name}
      </Name>

      <InfoItem>
        <strong>birth year:</strong>
        {birth_year}
      </InfoItem>

      <InfoItem>
        <strong>gender:</strong>
        {gender}
      </InfoItem>
    </div>
  );
};

ListDetails.propTypes = {
  details: PropTypes.shape({
    name: PropTypes.string.isRequired,
    birth_year: PropTypes.string.isRequired,
    gender: PropTypes.string.isRequired,
  }).isRequired,
};

export default ListDetails;
