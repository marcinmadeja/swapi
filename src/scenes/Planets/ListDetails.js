import React from 'react';
import {
  Name,
  InfoItem,
} from 'components/SwapiList/styles';

const ListDetails = ({ details }) => {
  const { name, climate, terrain, gravity } = details;

  return (
    <div>
      <Name>
        {name}
      </Name>

      <InfoItem>
        <strong>climate:</strong>
        {climate}
      </InfoItem>

      <InfoItem>
        <strong>terrain:</strong>
        {terrain}
      </InfoItem>

      <InfoItem>
        <strong>gravity:</strong>
        {gravity}
      </InfoItem>
    </div>
  );
};

export default ListDetails;
