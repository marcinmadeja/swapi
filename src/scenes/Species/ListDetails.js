import React from 'react';
import {
  Name,
  InfoItem,
} from 'components/SwapiList/styles';

const ListDetails = ({ details }) => {
  const { name, classification, language, designation } = details;

  return (
    <div>
      <Name>
        {name}
      </Name>

      <InfoItem>
        <strong>classification:</strong>
        {classification}
      </InfoItem>

      <InfoItem>
        <strong>language:</strong>
        {language}
      </InfoItem>

      <InfoItem>
        <strong>designation:</strong>
        {designation}
      </InfoItem>
    </div>
  );
};

export default ListDetails;
