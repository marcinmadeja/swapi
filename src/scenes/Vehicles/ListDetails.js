import React from 'react';
import {
  Name,
  InfoItem,
} from 'components/SwapiList/styles';

const ListDetails = ({ details }) => {
  const { name, model, manufacturer } = details;

  return (
    <div>
      <Name>
        {name}
      </Name>

      <InfoItem>
        <strong>model:</strong>
        {model}
      </InfoItem>

      <InfoItem>
        <strong>manufacturer:</strong>
        {manufacturer}
      </InfoItem>
    </div>
  );
};

export default ListDetails;
