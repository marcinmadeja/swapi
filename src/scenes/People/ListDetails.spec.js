import React from 'react';
import 'jest-styled-components';
import ListDetails from './ListDetails';

const details = {
  name: 'Name',
  birth_year: '2001',
  gender: 'male',
};

const defaultProps = {
  details,
};

const setup = (props = {}, children = null) => {
  props = { ...defaultProps, ...props };
  const component = <ListDetails {...props}>{children}</ListDetails>;
  const shallowComponent = shallow(component);

  return {
    component,
    shallowComponent,
  };
};

describe('ListDetails', () => {
  it('should render correctly', () => {
    const { shallowComponent } = setup();
    expect(shallowComponent.exists()).toBe(true);
  });

  it('should match snapshot with passed details', () => {
    const { component } = setup();
    const tree = renderer
      .create(component)
      .toJSON();

    expect(tree).toMatchSnapshot(tree);
  });
});
