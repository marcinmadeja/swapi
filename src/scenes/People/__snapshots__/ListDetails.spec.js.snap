// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[object Object] 1`] = `
.c0 {
  margin: 0 0 21px;
  font-size: 20px;
  font-weight: 700;
}

.c1 {
  margin-bottom: 20px;
  -webkit-letter-spacing: 0.7px;
  -moz-letter-spacing: 0.7px;
  -ms-letter-spacing: 0.7px;
  letter-spacing: 0.7px;
}

.c1 strong {
  margin-bottom: 5px;
  display: block;
  font-size: 16px;
  text-transform: uppercase;
  -webkit-letter-spacing: initial;
  -moz-letter-spacing: initial;
  -ms-letter-spacing: initial;
  letter-spacing: initial;
}

<div>
  <h3
    className="c0"
  >
    Name
  </h3>
  <div
    className="c1"
  >
    <strong>
      birth year:
    </strong>
    2001
  </div>
  <div
    className="c1"
  >
    <strong>
      gender:
    </strong>
    male
  </div>
</div>
`;
